lib/docker-toolbelt.coffee

```diff
@@ -42,45 +49,111 @@ createChainIdFromParent = (parent, dgsts) ->
 
 	return createChainIdFromParent(dgst, dgsts[1..])
 
+getDiffIds = (dkroot, driver, imageId) ->
+	Promise.try ->
+		[ hashType, hash ] = imageId.split(':')
+		fs.readFileAsync(path.join(dkroot, "image/#{driver}/imagedb/content", hashType, hash))
+	.then(JSON.parse)
+	.get('rootfs').get('diff_ids')
+
+getCacheId = (dkroot, driver, layerId) ->
+	Promise.try ->
```

Comment:
You can use `Promise.method (dkroot, driver, layerId) ->` which I think is slightly nicer (and maybe better performance fwiw?)